KB(Scheduler): many different colors for the appointments (#783)
telerik/blazor-docs: knowledge-base/scheduler-many-colors-for-appointments.md

---

> title: Many different colors for the appointments

Reviewer: ```suggestion
title: Dynamic appointment colors
```

> description: How to apply many different colors to the appointments

Reviewer: ```suggestion
description: How to apply many different colors to the appointments dynamically
```

type: how-to

> page_title: Many different colors for the appointments

Reviewer: ```suggestion
page_title: Dynamic appointment colors
```

slug: scheduler-kb-many-colors-for-appointments
position: 
tags: scheduler, many, colors, appointment
ticketid: 1553828
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Scheduler for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

> I want to set different background color to the appointments. I see that this can be achieved through the [OnItemRender]({%slug scheduler-events%}#itemrender) event by adding custom CSS class to the appointment. However, this isn't a practical use case for us, as we will be applying many different styles, which can change depending on who is using the application, so we don't want to have that many classes for each case. Is there any other way to achieve the desired result?

Reviewer: ```suggestion
I want to set multiple background colors to the appointments dynamically. I can add custom CSS classes through the [OnItemRender]({%slug scheduler-events%}#itemrender) event. However, this isn't a practical use case for us, as we will be applying many different styles, which can change depending on who is using the application, so we don't want to have that many classes for each case. Is there any other way to achieve the desired result? We already have the colors in the database.
```

## Solution

> You can add background-color and color properties to the appointment model which will be used to provide the desired background and font color for each appointment. Use [Appointment Templates]({%slug scheduler-templates-appointment%}), wrap the appointment content in a container and set some inline styles (background-color, color) based on the context. The example below demonstrates the suggested approach.

Reviewer: Use shorter sentences or even a bullet list of step-by-step instructions.

````CSHTML
<style>

> /* remove the default padding, so the template container can be expanded to cover the whole appointment */

Reviewer: ```suggestion
/* remove the default padding, so the template container expands to cover the whole appointment */
```

    .k-scheduler .k-event{

>         padding:0px;

Reviewer: ```suggestion
        padding: 0px;
```

    } 
/* set some padding to the template containers, expand them and add any other desired customizations. 

> Background and font color should be set inline in the templates, so we can get their values from the context 

Reviewer: ```suggestion
Styles from the model will be obtained from the ItemTemplate context and set inline.
```

based on the specific appointment*/
    .custom-appointment{    
        width:100%;
        height:100%;
        padding:5px;        
    }
</style>

<TelerikScheduler Data="@Appointments"
                  @bind-Date="@StartDate" 
                  @bind-View="@CurrView" 
                  Height="600px" 
                  Width="800px">
    <ItemTemplate>
        @{
            var appt = context as SchedulerAppointment;
            <div class="custom-appointment"

>                  style="background-color:@appt.BackgroundColor; color:@appt.Color">

Reviewer: ```suggestion
                 style="background-color: @appt.BackgroundColor; color: @appt.Color">
```

                @appt.Title
            </div>
        }
    </ItemTemplate>
    <AllDayItemTemplate>
        @{
            var appt = context as SchedulerAppointment;            
            <div class="custom-appointment"

>                  style="background-color:@appt.BackgroundColor; color:@appt.Color">

Reviewer: ```suggestion
                 style="background-color: @appt.BackgroundColor; color: @appt.Color">
```
